src/Content/PageInfo.php

```diff
@@ -253,10 +253,15 @@ public static function getRelevantUrlFromBody(string $body, bool $searchNakedUrl
 		// Fix for Mastodon where the mentions are in a different format
 		$body = preg_replace("~\[url=($URLSearchString)]([#!@])(.*?)\[/url]~is", '$2[url=$1]$3[/url]', $body);
 
-		preg_match("~(?<![!#@])\[url]($URLSearchString)\[/url]$~is", $body, $matches);
+		// Remove all hashtags and mentions
+		$body = preg_replace("/([#@!])\[url\=(.*?)\](.*?)\[\/url\]/ism", '', $body);
+
+		// Search for pure links
+		preg_match("/\[url\](https?:.*?)\[\/url\]/ism", $body, $matches);
```

Comment:
Are there any new "expected" and "body" cases we could add here:
https://github.com/friendica/friendica/blob/develop/tests/src/Content/PageInfoTest.php#L77 ?

Reply:
What about that (idk if this is the specific problem you had):
```php
			'bug-10196-pure-links' => [
				'expected' => 'https://text.com/with/media,
				'body' => '[url]http://test.com/with/media[/url]',
			],
```

Reply:
I just had the problem that some links hadn't been detected.